src/main/cpp/src/parse_uri.cu

```diff
@@ -990,25 +1040,30 @@ std::unique_ptr<cudf::column> parse_uri_to_query(cudf::strings_column_view const
   auto d_scalar = make_string_scalar(query_match, stream);
   auto col      = make_column_from_scalar(*d_scalar, 1);
 
+  if (ansi_mode) { detail::validate_input_uris(input, stream); }
```

Comment:
Let's move this validation step up, before creating other intermediate results, since that can be constly and we can short circuit other computation if an exception being thrown.

Reply:
Done. PTAL